junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -129,19 +132,18 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        // Commented because of issue https://github.com/k3po/k3po/issues/391
-//        prepared.countDown();
-//        startable.countDown();
-//        finished.countDown();
-        if (testThread != null) {
-            if (testThread != Thread.currentThread())
-                testThread.interrupt();
-        }
+        interruptTestThread();
```

Comment:
Maybe we can use a check on Thread.getState() to check if it is in an await state. What do you think ?